data/tutorials/language/0lg_00_basic_datatypes.md

```diff
@@ -0,0 +1,808 @@
+```ocaml
+# Ok 42;;
+- : (int, 'a) result = Ok 42
+
+# Error "Sorry";;
+- : ('a, string) result = Error "Sorry"
+```
+
+Operations on results are provided by the [`Result`](/api/Result.html) module. Results are discussed in the [Error Handling](/docs/error-handling) guide.
+
+### Tuples
+
+Here is a tuple containing two values, also known as a pair.
+```ocaml
+# (3, 'a');;
+- : int * char = (3, 'a')
+```
+
+That pair contains the integer `3` and the character `'a'`; its type is `int * char`. The `*` symbol stands for _product type_.
+
+This generalises to tuples with 3 or more elements. For instance, `(6.28, true, "hello")` has type `float * bool * string`. The types `int * char` and `float * bool * string` are called _product types_. The `*` symbol is used to denote types bundled together in products.
+
+The predefined function `fst` returns the first element of a pair, while `snd` returns the second element of a pair.
+```ocaml
+# fst (3, 'a');;
+- : int = 3
+
+# snd (3, 'a');;
+- : char = 'a'
+```
+
+In the standard library, both are defined using pattern matching. Here is how a function extracts the third element of the product of four types:
+```ocaml
+# let f x = match x with (a, b, c, d) -> c;;
+val f : 'a * 'b * 'c * 'd -> 'c = <fun>
+```
+
+Note that types `int * char * bool`, `int * (char * bool)`, and `(int * char) * bool` are not the same. The values `(42, 'a', true)`, `(42, ('a', true))`, and `((42, 'a'), true)` are not equal. In mathematical language, the product type operator `*` is not associative.
+
+### Functions
+
+The type of functions from type `a` to type `b` is written `a -> b`. Here are a few examples:
+```ocaml
+# fun x -> x * x;;
+- : int -> int = <fun>
+
+# (fun x -> x * x) 9;;
+- : int = 81
+```
+
+The first expression is an anonymous function of type `int -> int`. The type is inferred from the expression `x * x`, which must be of type `int` since `*` is an operator that returns an `int`. The `<fun>` printed in place of the value is a token, meaning functions don't have a value to be displayed. This is because, if they have been compiled, their code is no longer available.
+
+The second expression is function application. The parameter `9` is applied, and the result `81` is returned.
+
+```ocaml
+# fun x -> x;;
+- : 'a -> 'a = <fun>
+
+# (fun x -> x) 42;;
+- : int = 42
+
+# (fun x -> x) "This is really disco!";;
+- : string = "This is really disco!"
+```
+
+The first expression is another anonymous function. It is the identity function, it can be applied to anything, and it returns its argument unchanged. This means that its parameter can be of any type, and its result has the same type. The same code can be applied to data of different types. This is called _polymorphism_.
```

Comment:
```suggestion
The first expression is another anonymous function. It is the _identity_ function, it can be applied to anything, and it returns its argument unchanged. This means that its parameter can be of any type, and its result has the same type. The same code can be applied to data of different types. This is called _polymorphism_.
```

This use of the math term “identity”